protocol/src/types/service_context.rs

```diff
@@ -96,12 +96,11 @@ impl ServiceContext {
         self.events.borrow().clone()
     }
 
-    pub fn sub_cycles(&self, cycles: u64) -> ProtocolResult<()> {
+    pub fn sub_cycles(&self, cycles: u64) {
         if self.get_cycles_used() + cycles <= self.cycles_limit {
             *self.cycles_used.borrow_mut() = self.get_cycles_used() + cycles;
-            Ok(())
         } else {
-            Err(ServiceContextError::OutOfCycles.into())
+            panic!("out of cycles");
```

Comment:
This should be considered as a failed tx instead of a system error.

Reply:
Fixed. see:
```rust
    TokenStream::from(quote! {
        #func_vis fn #func_name#generics(#inputs) #ret {
            if !#request_ident.sub_cycles(#cycles_value) {
                return ServiceResponse::<_>::from_error(2, "cycles macro consume cycles fialed: out of cycles");
            }
            #body
        }
    })
```